### ISSUE TITLE
[BUG] New replicas are always stopped and show N/A

### ISSUE THREAD

**Issue by manuel-koch** [longhorn/longhorn#9413]
## Describe the bug

Scheduling more replicas for multiple volumes in the cluster are not possible.
When ever I increase the replica count or "delete" one of the strange stopped replicas it pops up again in this
stopped / N/A state.

![longhorn_replica_stopped_and_na](https://github.com/user-attachments/assets/ed82e3bc-bef8-4ed0-bdb7-0cf04209c922)

## To Reproduce

Don't know how to reproduce the issue. May be related to the fact that our cluster lost one node temporarily and that longhorn may have failed to sync properly afterwards.

## Expected behavior

Updating the replica count always results in the given number of running replicas.

## Support bundle for troubleshooting

supportbundle_0223e13d-568a-42b0-95a2-433d639e321c_2024-09-06T09-50-17Z.zip sent to longhorn-support-bundle@suse.com

## Environment

 - Longhorn version: 1.4.2 ( later upgraded to 1.6.2 exhibiting same problem, see below )
 - Impacted volume (PV): 
      - pvc-eaaa20cd-e28f-43b8-ae5e-2e7860c98d6b
      - pvc-a2ec206b-4de1-4691-8e8a-04c01f81f8fd
      - pvc-eaaa20cd-e28f-43b8-ae5e-2e7860c98d6b
 - Installation method: Rancher Catalog App
 - Kubernetes distro and version: k3s, v1.26.4
   - Total number of nodes in cluster: 6
   - Number of control plane nodes in the cluster: 3 ( are workers too )
   - Number of worker nodes in the cluster: 6
 - Node config
   - OS type and version: 
   - Kernel version:
   - CPU per node:
   - Memory per node:
   - Disk type: HDD
   - Network bandwidth between the nodes (Gbps):
 - Underlying Infrastructure: Baremetal
 - Number of Longhorn volumes in the cluster: 6

**Comment by manuel-koch** [comment]
Might be unrelated, but I see loggings from longhorn that seem to indicate problems:
```
longhorn-system/longhorn-manager-jsntb[longhorn-manager]: time="2024-09-06T14:32:07Z" level=debug msg="CheckEngineImageReadiness: nodes [agent-3] don't have the engine image rancher/mirrored-longhornio-longhorn-engine:v1.4.2"

longhorn-system/longhorn-manager-jsntb[longhorn-manager]: time="2024-09-06T14:32:07Z" level=error msg="There's no available disk for replica pvc-eaaa20cd-e28f-43b8-ae5e-2e7860c98d6b-r-8ada73e3, size 8589934592"
longhorn-system/longhorn-manager-jsntb[longhorn-manager]: time="2024-09-06T14:32:07Z" level=error msg="unable to schedule replica" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=agent-4 owner=agent-4 replica=pvc-eaaa20cd-e28f-43b8-ae5e-2e7860c98d6b-r-8ada73e3 state=attached volume=pvc-eaaa20cd-e28f-43b8-ae5e-2e7860c98d6b
longhorn-system/longhorn-manager-jsntb[longhorn-manager]: time="2024-09-06T14:32:07Z" level=debug msg="Found 0 node has at least one schedulable disk" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=agent-4 owner=agent-4 state=attached volume=pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289
```

**Comment by manuel-koch** [comment]
Upgraded Longhorn in my cluster from v1.4.2 to v1.6.2.

Looks like the problem is still happening when editing the number of replicas.
E.g. for one of the volumes I changed replica count from 2 to 3 and got that "stopped" / "N/A" problem immediately again:
![Bildschirmfoto 2024-09-09 um 14 58 29](https://github.com/user-attachments/assets/a11b8a2b-8370-4b5c-97de-7b014d42e5b7)

**Comment by manuel-koch** [comment]
I see logs in longhorn namespace like the following:
```
longhorn-system/longhorn-manager-c97s5[longhorn-manager]: time="2024-09-09T13:02:48Z" level=error msg="There's no available disk for replica pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289-r-eea546a8, size 1073741824" func="scheduler.(*ReplicaScheduler).ScheduleReplica" file="replica_scheduler.go:101"
longhorn-system/longhorn-manager-c97s5[longhorn-manager]: time="2024-09-09T13:02:48Z" level=warning msg="Failed to schedule replica" func="controller.(*VolumeController).reconcileVolumeCondition" file="volume_controller.go:1769" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=agent-4 owner=agent-4 replica=pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289-r-eea546a8 state=attached volume=pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289
longhorn-system/longhorn-manager-c97s5[longhorn-manager]: time="2024-09-09T13:02:48Z" level=error msg="There's no available disk for replica pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289-r-eea546a8, size 1073741824" func="scheduler.(*ReplicaScheduler).ScheduleReplica" file="replica_scheduler.go:101"
longhorn-system/longhorn-manager-c97s5[longhorn-manager]: time="2024-09-09T13:02:48Z" level=warning msg="Failed to schedule replica" func="controller.(*VolumeController).reconcileVolumeCondition" file="volume_controller.go:1769" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=agent-4 owner=agent-4 replica=pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289-r-eea546a8 state=attached volume=pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289
```

**Comment by derekbit** [comment]
> There's no available disk for replica

So, can you check why there is no available disk? disk space or other factors?

**Comment by manuel-koch** [comment]
Looks like there is still space left on that cluster agent/node (agent-4):
```
Free space on filesystems:
Filesystem                                              Size  Used Avail Use% Mounted on
udev                                                    3.8G     0  3.8G   0% /dev
/dev/sda1                                                76G   35G   38G  48% /
/dev/longhorn/pvc-11e4195e-3315-4082-994b-2ffe0de0cc94  2.0G  100M  1.8G   6% /var/lib/kubelet/pods/fc76ac54-689f-45ee-812f-c360683a35b8/volumes/kubernetes.io~csi/pvc-11e4195e-3315-4082-994b-2ffe0de0cc94/mount
/dev/longhorn/pvc-1569853e-c1da-4f97-9b83-89ee76c6cb05  2.0G  269M  1.7G  14% /var/lib/kubelet/pods/773dad1e-06b8-4003-8644-55f0f281f811/volumes/kubernetes.io~csi/pvc-1569853e-c1da-4f97-9b83-89ee76c6cb05/mount
/dev/longhorn/pvc-a2ec206b-4de1-4691-8e8a-04c01f81f8fd  974M  664M  295M  70% /var/lib/kubelet/pods/5b05cf9f-54bc-4612-93da-b72daf54a427/volumes/kubernetes.io~csi/pvc-a2ec206b-4de1-4691-8e8a-04c01f81f8fd/mount
/dev/longhorn/pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289  974M  743M  215M  78% /var/lib/kubelet/pods/8eb0fe16-93c2-40bd-8d96-f71cb1921e2f/volumes/kubernetes.io~csi/pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289/mount
```

@derekbit Do you have any suggestion of commands I should run to get more info related to my problem ?

**Comment by derekbit** [comment]
@manuel-koch The above information doesn't include `nodes.longhorn.io`. Can you provide a support bundle? Thanks.

**Comment by derekbit** [comment]
@manuel-koch 
longhorn-manager emits the error
```
2024-09-09T12:59:18.158369683Z time="2024-09-09T12:59:18Z" level=error msg="There's no available disk for replica pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289-r-eea546a8, size 1073741824" func="scheduler.(*ReplicaScheduler).ScheduleReplica" file="replica_scheduler.go:101"
```
The disk space looks enough and no idea on top of my head for now. 
@PhanLe1010 @ejweber Any thoughts?

**Comment by ejweber** [comment]
I ran the numbers on disk `47b2fd162eb346f4` for node `agent-1`, and it definitely has space for `pvc-fc125f83-3c5c-4078-9b02-cfaa2ced6289-r-eea546a8`.

For now, the only thing I can think of is that https://longhorn.io/docs/1.6.2/references/settings/#replica-zone-level-soft-anti-affinity is set to `false`, when we default it to `true`. It doesn't look there are any zones set up in this cluster, but maybe the scheduler is treating the entire cluster as one zone and refusing to schedule additional replicas to it?

Are you intentionally running with `replica-zone-level-soft-antif-affinity == false` @manuel-koch? If not, can you try changing it to `true`?

**Comment by ejweber** [comment]
I was able to reproduce this on `master-head` by:

- Create a volume with one replica in a cluster that uses no zones.
- Set `replica-zone-soft-anti-affinity = false`.
- Attempt to increase the replica count to two.

On `master-head`, we don't actually create the replica due to some new pre-check logic, but the effect is largely the same. We have a replica scheduling failure.

```
[longhorn-manager-ghn6h longhorn-manager] time="2024-09-09T21:20:34Z" level=warning msg="Unable to create new replica test-r-47520ea4" func="controller.(*VolumeController).replenishReplicas" file="volume_controller.go:2305" accessMode=rwo controller=longhorn-volume error="No available disk candidates to create a new replica of size 5368709120" frontend=blockdev migratable=false node=eweber-v126-worker-9c1451b4-6464j owner=eweber-v126-worker-9c1451b4-6464j state=attached volume=test
```

I'm pretty confident scheduling will work again after you update the setting @manuel-koch. IMO this is not a bug. The user explicitly asked Longhorn not to schedule multiple replicas for a volume in the same zone, so Longhorn responded with a replica scheduling failure. This behavior prevents the user from accidentally running with less redundancy than intended.

**Comment by manuel-koch** [comment]
I have not intentionally edited this particular setting `replica-zone-soft-anti-affinity = false`, thus might be there from the beginning or a left-over from incomplete migration during upgrade.
The description of this setting in UI clearly states `Nodes don't belong to any Zone will be treated as in the same Zone.`.

Given that I have not edited the setting, it would have been nice if the UI or at least the logs would have clearly identified the problem and gave a useful feedback to user.

@ejweber I can confirm that I have not configured any zones for the nodes and that the particular 3rd replica I requested ( see screenshot above ) was scheduled successfully after I switched to `replica-zone-soft-anti-affinity = true`.

@ejweber @derekbit Thank you for your help and the fast feedback !
